src/modules/Channel/components/SuggestedMentionList/index.tsx

```diff
@@ -93,43 +108,75 @@ function SuggestedMentionList(props: SuggestedMentionListProps): JSX.Element {
     }
   }, [inputEvent]);
 
-  /* Fetch member list */
+  /* Update member list for non-super channels */
   useEffect(() => {
-    if (!channelInstance?.createMemberListQuery) {
-      logger.warning('SuggestedMentionList: Creating member list query failed');
+    if (lastSearchString && searchString.indexOf(lastSearchString) === 0 && currentMemberList.length === 0) {
+      // Don't need to request query again
       return;
     }
+    // Add member list query for customization
+    if (!channelInstance.isSuper) {
+      const suggestingMembers = getCurrentSearchedMemberList();
+      if (suggestingMembers.length < 1) {
+        logger.info('SuggestedMentionList: Channel\'s member list is empty');
+      } else {
+        logger.info('SuggestedMentionList: Channel\'s member list has been updated.', { memberList: suggestingMembers });
+        setCurrentUser(suggestingMembers[0]);
+      }
+      setLastSearchString(searchString);
+      onFetchUsers(suggestingMembers);
+      setCurrentMemberList(suggestingMembers);
+    }
+  }, [
+    channelInstance?.url,
+    channelInstance.members.length,
+    // We have to be specific like this or React would not recognize the changes in instances.
+    channelInstance.members.map((member: Member) => member.nickname).join(),
+    channelInstance.members.map((member: Member) => member.state).join(),
```

Comment:
You can define the CurrentSearchedMemberList above and put it here instead of these